### PR TITLE
Add OutlierRemover and tests

### LINKED ISSUE #121: [FEATURE] OutlierRemover Transformer
Since we're going to be having multiple outlier removal models we might want to make a special type of transformer that will remove outliers from a pipeline. 

```python
pipeline = Pipeline([
    ....
    ("remove_outliers", OutlierRemoval(GMMDetector(...)),
    ...
])
```

## Base Behavior 

The `OutlierRemoval` will filter out any row where the model passed to the `OutlierRemoval` transformer does not return 0. This allows for a pre-trained model to be applied but it also allows for a outlier-detector model to be trained internally.

### PR DESCRIPTION
**Pull request by jcshoekstra**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes #121 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] My code follows the style guidelines (flake8)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (also to the readme.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added tests to check whether the new feature adheres to the sklearn convention
- [x] New and existing unit tests pass locally with my changes

If you feel your PR is ready for a review, ping @koaning or @mbrouns

### CODE REVIEW

**Comment by jcshoekstra** [comment]
@koaning

File: tests/test_meta/test_outlier_remover.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    estimator_checks.check_fit2d_1sample,
+    estimator_checks.check_fit2d_1feature,
+    estimator_checks.check_fit1d,
+    estimator_checks.check_get_params_invariance,
+    estimator_checks.check_set_params,
+    estimator_checks.check_dont_overwrite_parameters,
+    estimator_checks.check_transformers_unfitted
+]))
+def test_estimator_checks(test_fn):
+    outlier_remover = OutlierRemover(outlier_detector=GMMOutlierDetector(), refit=True)
```
Comment (koaning) [inline tests/test_meta/test_outlier_remover.py]: might be nice to also add `IsolationForest` here as well from normal `sklearn`.

File: tests/test_meta/test_outlier_remover.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+def test_no_outliers(mock_outlier_detector, mocker):
+    mock_outlier_detector.fit.return_value = None
+    mock_outlier_detector.predict.return_value = np.array([[1, 1]])
+    mocker.patch('sklego.meta.clone').return_value = mock_outlier_detector
+
+    outlier_remover = OutlierRemover(outlier_detector=mock_outlier_detector, refit=True)
+    outlier_remover.fit(X=np.array([[1, 1], [2, 2]]))
+    assert len(outlier_remover.transform_train(np.array([[1, 1], [2, 2]]))) == 2
+
+
```
Comment (koaning) [inline tests/test_meta/test_outlier_remover.py]: - might be nice to add one more test that demonstrates that `refit=False` also works.

Comment (jcshoekstra) [reply tests/test_meta/test_outlier_remover.py]: Agreed! I have added a test for this.

File: tests/test_meta/test_outlier_remover.py (hunk 1 above)
Comment (jcshoekstra) [reply tests/test_meta/test_outlier_remover.py]: Sure!

**Pull request merged**